setup.go

```diff
@@ -118,22 +139,124 @@ func Setup(ctx context.Context, cfg *Config) (*Node, error) {
 
 	bsctx := metri.CtxScope(ctx, "rainbow")
 
-	r1, err := routingv1client.New(cfg.RoutingV1, routingv1client.WithStreamResultsRequired())
-	if err != nil {
-		return nil, err
-	}
-
-	vs := httpcontentrouter.NewContentRoutingClient(r1)
+	var pr routing.PeerRouting
+	var vs routing.ValueStore
+	var cr routing.ContentRouting
 
 	opts = append(opts, libp2p.Routing(func(h host.Host) (routing.PeerRouting, error) {
-		return vs, nil
+		if cfg.RoutingV1 != "" {
+			routingClient, err := delegatedHTTPContentRouter(cfg.RoutingV1, routingv1client.WithStreamResultsRequired())
+			if err != nil {
+				return nil, err
+			}
+			pr = routingClient
+			vs = routingClient
+			cr = routingClient
+		} else {
+			// If there are no delegated routing endpoints run an accelerated Amino DHT client and send IPNI requests to cid.contact
+
+			// TODO: This datastore shouldn't end up containing anything anyway so this could potentially just be a null datastore
+			memDS, err := levelds.NewDatastore("", nil)
+			if err != nil {
+				return nil, err
+			}
+
+			var dhtHost host.Host
+			if cfg.DHTSharedHost {
+				dhtHost = h
+			} else {
+				dhtHost, err = libp2p.New(
+					libp2p.NoListenAddrs,
+					libp2p.BandwidthReporter(bwc),
+					libp2p.DefaultTransports,
+					libp2p.DefaultMuxers,
+				)
+				if err != nil {
+					return nil, err
+				}
+			}
+
+			var standardClient *dht.IpfsDHT
+			var fullRTClient *fullrt.FullRT
+
+			if cfg.DHTType == Combined || cfg.DHTType == Standard {
+				standardClient, err = dht.New(ctx, dhtHost,
+					dht.Datastore(memDS),
+					dht.BootstrapPeers(dht.GetDefaultBootstrapPeerAddrInfos()...),
+					dht.Mode(dht.ModeClient),
+				)
+				if err != nil {
+					return nil, err
+				}
+			}
+
+			if cfg.DHTType == Combined || cfg.DHTType == Accelerated {
+				fullRTClient, err = fullrt.NewFullRT(dhtHost, dht.DefaultPrefix,
+					fullrt.DHTOption(
+						dht.Validator(record.NamespacedValidator{
+							"pk":   record.PublicKeyValidator{},
+							"ipns": ipns.Validator{KeyBook: h.Peerstore()},
+						}),
+						dht.Datastore(memDS),
+						dht.BootstrapPeers(dht.GetDefaultBootstrapPeerAddrInfos()...),
+						dht.BucketSize(20),
+					))
+				if err != nil {
+					return nil, err
+				}
+			}
+
+			var dhtRouter routing.Routing
+			switch cfg.DHTType {
+			case Combined:
+				dhtRouter = &bundledDHT{
+					standard: standardClient,
+					fullRT:   fullRTClient,
+				}
+			case Standard:
+				dhtRouter = standardClient
+			case Accelerated:
+				dhtRouter = fullRTClient
+			default:
+				return nil, fmt.Errorf("unsupported DHT type")
+			}
```

Comment:
I think I regular DHT should just use combined.

Reply:
Yeah, feel free to rename however you want. Just wanted to illustrate the options for you and allow us to test them out if we want to evaluate the performance of them.